corehq/apps/users/views/__init__.py

```diff
@@ -427,6 +441,25 @@ def post(self, request, *args, **kwargs):
         if self.request.is_view_only:
             return self.get(request, *args, **kwargs)
 
+        if (toggles.ENTERPRISE_SSO.enabled_for_request(self.request)
+                and self.request.POST['form_type'] == 'trust-identity-provider'):
+            idp = IdentityProvider.get_active_identity_provider_by_username(
+                self.editable_user.username
+            )
+            if idp:
+                idp.create_trust_with_domain(
+                    self.domain,
+                    self.request.user.username
+                )
```

Comment:
Is there an easy way for the user to undo this action, for example if they did it by accident?

Reply:
this is a good point. will include this